pulser-core/pulser/sequence/sequence.py

```diff
@@ -553,6 +553,8 @@ def check_retarget(ch_obj: Channel) -> bool:
                         continue
                 if not strict:
                     channel_match[old_ch_name] = new_ch_id
+                    if ch_match_err.startswith(base_msg):
+                        ch_match_err = ""
```

Comment:
nit: You could a comment like
```suggestion
                    # Found a match, clear match error msg for this
                    # this channel
                    if ch_match_err.startswith(base_msg):
                        ch_match_err = ""
```